features/FEATURE_BLE/targets/TARGET_CORDIO/source/CordioBLE.cpp

```diff
@@ -219,10 +219,14 @@ const SecurityManager& BLE::getSecurityManager() const
 {
     static pal::MemorySecurityDb m_db;
     pal::vendor::cordio::CordioSecurityManager &m_pal = pal::vendor::cordio::CordioSecurityManager::get_security_manager();
+
+    BLE *self = const_cast<BLE*>(this);
```

Comment:
Could you unify access to generic gap and generic gatt client ? 

I advise you to let `getGattClient` return a `GenericGattClient` so there's no static cast required. `const_cast` can remain explicit.

Reply:
Thanks, it looks better. As covariant return type are allowed in C++; it would be a good thing if we return the most derived type in overridden virtual functions.